agc/anal.c

```diff
@@ -0,0 +1,166 @@
+#include <stdbool.h>
+#include <stdio.h>
+#include <r_anal.h>
+#include "anal.h"
+
+void analyze_insn(RAnalOp *op, unsigned int address, ut16 value, bool shift) {
+    // extracode flag saved between invocations
+    static bool extra_opcode_bit = false;
+
+    // distinguish between RW and RO memory
+    bool erasable = false;
+    if (address < 02000)
+        erasable = true;
+
+    // shift data one bit to the right if we work with memory as it was stored
+    // on the original AGC hardware.
+    if (shift) value <<= 1;
+
+    // default info
+    op->size = 2;
+    op->type = R_ANAL_OP_TYPE_UNK;
+
+    if (!extra_opcode_bit) {
```

Comment:
Use positive conditionals and move the body of each block into a separate function to ease reading